src/components/main/library/library.js

```diff
@@ -1,7 +1,117 @@
-import React from "react";
+import React, { useState } from "react";
+import { Box, Button, Text, Heading, Container, Flex } from "@chakra-ui/react";
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  TableCaption,
+  TableContainer,
+} from "@chakra-ui/react";
+import { DeleteIcon } from "@chakra-ui/icons";
+import { v4 as uuidv4 } from "uuid";
 
 const Library = () => {
-  return <div>library</div>;
+  const [books, setBooks] = useState([]);
+
+  const AreBooksEmpty = () => {
+    if (books.length === 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const AddBook = () => {
+    setBooks([
+      ...books,
+      {
+        key: uuidv4(),
+        author: "Colette",
+        title: "Green Sealing Wax",
+        form: ["Fiction", "Short Story"],
+        topics: ["Bildungsroman", "Coming of age"],
+        historicalEra: "Modern",
+      },
+    ]);
+  };
+
+  const DeleteLastBook = () => {
+    setBooks((prevBooks) => {
+      const updatedBooks = prevBooks.slice(0, -1);
+      return updatedBooks;
+    });
+  };
+
+  const DeleteSpecificBook = (id) => {
+    const updatedBooks = books.filter((book) => book.key !== id);
+    setBooks(updatedBooks);
+  };
+
+  if (AreBooksEmpty()) {
+    return (
+      <Box>
+        <Heading>Library</Heading>
+        <Text>You haven&apos;t added any books! Click below to add a book</Text>
+        <Button onClick={AddBook}>Add a book!</Button>
+      </Box>
+    );
+  }
+  return (
+    <Box>
+      Library
+      <Button onClick={AddBook}>Add a book!</Button>
+      <Button onClick={DeleteLastBook}>Remove last book!</Button>
+      <Container maxW="90%">
+        <TableContainer>
+          <Table variant="simple" size={{ base: "sm", md: "md", lg: "lg" }}>
+            <TableCaption>Your main library</TableCaption>
+            <Thead>
+              <Tr>
+                <Th>Author</Th>
+                <Th>Title</Th>
+                <Th display={{ base: "none", md: "table-cell" }}>Form</Th>
+                <Th display={{ base: "none", md: "table-cell" }}>Topics</Th>
+                <Th display={{ base: "none", md: "table-cell" }}>
+                  Historical Era
+                </Th>
+                <Th>Actions</Th>
+              </Tr>
+            </Thead>
+            <Tbody>
+              {books.map((book) => {
+                return (
+                  <Tr key={book.key} _hover={{ background: "blue.200" }}>
```

Comment:
```suggestion
                  <Tr key={book.id} _hover={{ background: "blue.200" }}>
```
_if you change the book.key to book.id_